server/src/main/java/org/elasticsearch/cluster/routing/allocation/shards/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -439,10 +458,11 @@ public class ShardAllocationCounts {
         final Map<Diagnosis.Definition, Set<String>> diagnosisDefinitions = new HashMap<>();
 
         public void increment(ShardRouting routing, ClusterState state, NodesShutdownMetadata shutdowns, boolean verbose) {
+            final TimeValue replicaUnassignedThreshold = clusterService.getClusterSettings().get(REPLICA_UNASSIGNED_BUFFER_TIME);
```

Comment:
This is different than the normal setup for dynamic cluster settings. Normally we have a class variable and a

```java
clusterService.getClusterSettings().addSettingsUpdateConsumer(REPLICA_UNASSIGNED_BUFFER_TIME, this::updateTheVar);
```

I think we want to do it with the update consumer, since that's the usual way for handling these settings